druid/src/box_constraints.rs

```diff
@@ -270,6 +270,16 @@ impl BoxConstraints {
             }
         }
     }
+
+    /// Set max width to max_width
+    pub fn set_max_width(&mut self, max_width: f64) {
+        self.max.width = max_width;
+    }
+
+    /// Set max height to max_height
+    pub fn set_max_height(&mut self, max_height: f64) {
+        self.max.height = max_height;
+    }
```

Comment:
^ made me notice there arent any other `&mut` in BC. I added those two, so it is easier for me to _just_ change maxes without changing anything else. I can of course create `Size`s and then create new BC but having the ability to just change maxes makes it so much cleaner.

Reply:
One way to have the convenience without breaking the mutability convention would be to have `BoxConstraints::with_max_width(&self, max_width) -> BoxConstraints` functions.

But I'm actually wondering if there's a better API that will also ensure that the constraints remain valid (because with `set_max_width` you could break the invariant that the minimum constraint is smaller than the maximum constraint). What about something like:
```rust
/// Sets the max on a given axis to infinity.
pub fn unbound_max(&self, axis: Axis) -> BoxConstraints;
// and maybe, for convenience
pub fn unbound_max_width(&self) -> BoxConstraints;
pub fn unbound_max_height(&self) -> BoxConstraints;

/// Shrinks the max dimension on the given axis.
/// (TODO: specify what should happen if this shrinks the max to less than the min. Decrease the min?
/// Just set the max to the min?)
pub fn shrink_max_to(&self, axis: Axis, dim: f64) -> BoxConstraints;
// and maybe, for convenience
pub fn shrink_max_width_to(&self, dim: f64) -> BoxConstraints;
pub fn shrink_max_height_to(&self, dim: f64) -> BoxConstraints;
```

Then instead of
```rust
    bc.set_max_width(f64::INFINITY);
    bc.set_max_height(available);
```
you can say
```rust
    let bc = bc.unbound_max_width().shrink_max_height_to(available);
```

Reply:
I like it. BC is nothing but `Sizes` that are cheap to clone. I was just being overly cautions of clones. 
I'll make the change.